packages/core/examples/webLoader/registerWebImageLoader.ts

```diff
@@ -202,11 +211,17 @@ function _loadImageIntoBuffer(
           // If we have a target buffer, write to that instead. This helps reduce memory duplication.
           const { arrayBuffer, offset, length } = options.targetBuffer;
 
+          // since the web image was loaded via canvas api it has rgba format
+          // below we convert it to rgb since the vtk image data can work
+          // only with rgb format
+
           // @ts-ignore
-          const pixelDataRGBA = image.getPixelData();
+          const pixelDataRGB = image.getPixelData();
```

Comment:
Why do you need yet another image data array?  You don't even appear to use targetArray, so can you throw it away entirely?    I don't get this logic here.

Reply:
targetArray is used to view the portion of the full volume at the offset and length, we need it to fill the volume

Reply:
Oh, I see what you are doing here - the target array is created from the passed in arrayBuffer instance, so it is a view onto the destination object and you are setting it.  Missed that and thought you were creating a new instance.
You can avoid the copy entirely by passing targetBuffer into getPixelData and creating the Uint8array view inside the getPixelData.  Then, that getPixelData has enough information to do a bounds check to see that nothing is being cut off, and to throw an exception if the provided array is too small.

Reply:
It is not about only volume viewport and target buffer, it should handle both cases (Stack and volume) so I don't want to modify the getPixelData since that correctly return the pixelData for stack viewport, does that makes sense?

Reply:
That is why I would put the logic into getPixelData, setting it up as an optional parameter.  The return value will be identical in both cases, just if you are passed an array buffer, it uses that array buffer (+ offsets info), otherwise create new data.  That prevents the copy operation entirely, while being 100% compatible with either call method.

Reply:
Ok applied it